dockers/docker-auditd/auditd_config_files/30-audisp-tacplus.rules

```diff
@@ -22,8 +22,8 @@
 # We assume here that user 1000 is the first local user, and therefore
 # should not be looked up for tacacs.  You may need to change this for
 # your local configuration
--a always,exit -F arch=b32 -S execve -S exit -S exit_group -F auid>1000 -F auid!=4294967295 -k tacplus
--a always,exit -F arch=b64 -S execve -S exit -S exit_group -F auid>1000 -F auid!=4294967295 -k tacplus
```

Comment:
>execve [](http://example.com/codeflow?start=30&length=6)

What is the purpose of this change? how necessary is this change?

Reply:
@liuh-80 could you also review this part?

Reply:
if we leave 30-audisp-tacplus.rules unchanged, auditd will match with rule 1 and rule 2, and stop there, it won’t reach rule 3 or 4, so nothing gets logged under the process_audit key.
 
1.-a always,exit -F arch=b32 -S exit,execve,exit_group -F auid>1000 -F auid!=-1 -F key=tacplus
2. -a always,exit -F arch=b64 -S execve,exit,exit_group -F auid>1000 -F auid!=-1 -F key=tacplus
3. -a always,exit -F arch=b64 -S execve -F auid>=1000 -F auid!=-1 -F key=process_audit -F key=tacplus
4. -a always,exit -F arch=b32 -S execve -F auid>=1000 -F auid!=-1 -F key=process_audit -F key=tacplus

Reply:
with SYSCALL execve, key=tacplus will be key=70726F636573735F617564697401746163706C7573 in log message, key=70726F636573735F617564697401746163706C7573 covers both process_audit and tacplus. with SYSCALL exit, exit_group, the behavior is unchanged